src/utils/update/psc.to.beneficial.owner.type.mapper.ts

```diff
@@ -0,0 +1,191 @@
+import { OfficeAddress } from "@companieshouse/api-sdk-node/dist/services/company-profile/types";
+import { CompanyPersonWithSignificantControl } from "@companieshouse/api-sdk-node/dist/services/company-psc/types";
+import { BeneficialOwnerGov } from "../../model/beneficial.owner.gov.model";
+import { BeneficialOwnerIndividual } from "../../model/beneficial.owner.individual.model";
+import { BeneficialOwnerOther } from "../../model/beneficial.owner.other.model";
+import { Address, InputDate, NatureOfControlType, yesNoResponse } from "../../model/data.types.model";
+
+export const mapPscToBeneficialOwnerTypeIndividual = (psc: CompanyPersonWithSignificantControl): BeneficialOwnerIndividual => {
+  const service_address = mapAddress(psc.address);
+  const address = mapAddress(undefined);
+
+  const result: BeneficialOwnerIndividual = {
+    id: psc.links?.self,
+    link: {
+      self: psc.links?.self
+    },
+    first_name: psc.nameElements?.forename,
+    last_name: psc.nameElements?.surname,
+    nationality: psc.nationality,
+    second_nationality: undefined,
+    date_of_birth: mapDateOfBirth(psc),
+    is_service_address_same_as_usual_residential_address: isSameAddress(service_address, address) ? yesNoResponse.Yes : yesNoResponse.No,
+    usual_residential_address: address,
+    service_address: service_address,
+    start_date: psc.notifiedOn as any as InputDate, // notified is type string not input date
+    // is_on_sanctions_list: psc.isSanctioned === "true" ? yesNoResponse.Yes : yesNoResponse.No,
+  };
+  mapNatureOfControl(psc, result, false);
+  return result;
+};
+
+export const mapPscToBeneficialOwnerOther = (psc: CompanyPersonWithSignificantControl): BeneficialOwnerOther => {
+  const service_address = mapAddress(psc.address);
+  const principal_address = mapAddress(psc.address);
+
+  const result: BeneficialOwnerOther = {
+    id: psc.links?.self,
+    link: {
+      self: psc.links?.self
+    },
+    name: psc.name,
+    principal_address: principal_address,
+    service_address: service_address,
+    is_service_address_same_as_principal_address: isSameAddress(service_address, principal_address) ? yesNoResponse.Yes : yesNoResponse.No,
+    legal_form: psc.identification?.legalForm,
+    law_governed: psc.identification?.legalAuthority,
+    public_register_name: psc.identification?.placeRegistered,
+    registration_number: psc.identification?.registrationNumber,
+    is_on_register_in_country_formed_in: undefined,
+    start_date: psc.notifiedOn as any as InputDate, // notified is type string not input date
```

Comment:
no need for two as?

Reply:
Needs conversion from string to InputDate.

Reply:
OK to set to undefined and do on second pass

Reply:
Done